jaxopt/_src/projection.py

```diff
@@ -419,6 +436,34 @@ def fun(alpha, cost_matrix, marginals_a, marginals_b):
   return fun
 
 
+def _make_dual(delta_vmap, gamma):
+  r"""Make the objective function of dual variables.
+
+  Args:
+    delta_vmap: The smoothed version of delta function, acting on each column of
+      its matrix-valued input.
+    gamma: A regularization constant.
+  Returns:
+    A cost function of dual variables. Cf. Equation (7) in
+      https://arxiv.org/abs/1710.06276
+  """
+
+  def fun(alpha_beta, cost_matrix, marginals_a, marginals_b):
+    len_alpha = len(marginals_a)
+    alpha = alpha_beta[:len_alpha]
+    beta = alpha_beta[len_alpha:]
+    alpha_column = alpha[:, jnp.newaxis]
+    beta_row = beta[jnp.newaxis, :]
+    # Make a dual constraint matrix, whose (i,j)-th entry is
+    # alpha[i] + beta[j] - c[i,j].
+    dual_constraint_matrix = alpha_column + beta_row - cost_matrix
+    delta_dual_constraints = delta_vmap(dual_constraint_matrix, gamma)
```

Comment:
Add a comment here "JAXopt solvers minimize functions hence the sign is the opposite of Eqn (7)"